### PR TITLE
Fetaure/bye bye super duper burger

### PR DESCRIPTION
**Pull request by 7illusions**

### CODE REVIEW

File: Emby.Mobile.Universal/Controls/HeaderControl.xaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            <ColumnDefinition Width="Auto" />
+            <ColumnDefinition Width="Auto" />
+        </Grid.ColumnDefinitions>
+        <Image Stretch="UniformToFill"
+               Margin="5"
+               Source="ms-appx:///Assets/LogoLight.png" />
+        <Button x:Name="SearchButton"
+                Style="{StaticResource EmptyButtonStyle}"
+                Grid.Column="2">
+            <TextBlock x:Name="SearchTextBlock"
```
Comment (ScottIsAFool) [inline Emby.Mobile.Universal/Controls/HeaderControl.xaml]: Just as a thing of habit, use FontIcon for this, there's already an MDL2 fonticonstyle where you can set the glyph

File: Emby.Mobile.Universal/Controls/HeaderControl.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    public sealed partial class HeaderControl : UserControl
+    {
+        public HeaderMenuViewModel Header => DataContext as HeaderMenuViewModel;
+        public HeaderControl()
+        {
+            InitializeComponent();
+            DataContextChanged += (s, e) =>
+            {
+                if (Header != null)
```
Comment (ScottIsAFool) [inline Emby.Mobile.Universal/Controls/HeaderControl.xaml.cs]: Why is this needed?

File: Emby.Mobile.ViewModels/ViewModels/HeaderMenuViewModel.cs [hunk 3]
```diff
@@ -25,15 +27,15 @@ public BurgerMenuViewModel(IServices services) : base(services)
                     Start();
                 };
 
-                Views = new ObservableCollection<ItemViewModel>();
+                SearchResults = new ObservableCollection<SearchHint>();
```
Comment (ScottIsAFool) [inline Emby.Mobile.ViewModels/ViewModels/HeaderMenuViewModel.cs]: This would be a SearchHintViewModel, but we'll come to that later on ;)

File: Emby.Mobile.ViewModels/ViewModels/HeaderMenuViewModel.cs [hunk 4]
```diff
@@ -95,29 +95,38 @@ private void SetUsernameAndProfilePicture()
             });
         }
 
-        private async Task LoadViews(bool isRefresh)
+        async void OnSearchTextChanged()
```
Comment (ScottIsAFool) [inline Emby.Mobile.ViewModels/ViewModels/HeaderMenuViewModel.cs]: private...

File: Emby.Mobile.Universal/Controls/HeaderControl.xaml.cs (hunk 2 above)
Comment (7illusions) [inline Emby.Mobile.Universal/Controls/HeaderControl.xaml.cs]: If DataContext is null when forcing an Bindings update, the app will crash.
